### ISSUE TITLE
Compile warning occurs while adapting my project to Elixir 1.18

### ISSUE THREAD

**Issue by stevencch99** [puid/Elixir#15]
Env:
- macOS 14.4.1 (23E224)
- Erlang/OTP 27 [erts-15.1.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit]
- Elixir 1.18.0 (compiled with Erlang/OTP 27)

LSP warning shown as below:

```elixir
[{
	"resource": "MY_APP/MODULE.ex",
	"owner": "_generated_diagnostic_collection_name_#5",
	"severity": 4,
	"message": "Range.new/2 and first..last default to a step of -1 when last < first. Use Range.new(first, last, -1) or first..last//-1, or pass 1 if that was your intention",
	"source": "Elixir",
	"startLineNumber": 2,
	"startColumn": 3,
	"endLineNumber": 2,
	"endColumn": 3
}]
```

**Comment by dingosky** [comment]
Hi Steven. Any way you could provide a bit more context? Is MY_APP/MODULE.ex a PUID generated module? I haven't seen this warning before, and the detail doesn't trigger any guesses of where to look on my part.  -Paul

**Comment by stevencch99** [comment]
Hi, the module is our product code, it using `Puid` with only one function like below.

```elixir
defmodule MyApp.ReferenceCode do
  use Puid, bits: 64, chars: :alphanum

  def gen_code(form_number) do
    "#{form_number} - #{generate()}"
    |> String.replace("-", "")
    |> String.replace("_", "")
  end
end
```

We have `{:puid, "~> 2.3"},` in our mix deps, and can see the warning message by `mix compile --warnings-as-errors`:
![CleanShot 2024-12-24 at 22 53 34](https://github.com/user-attachments/assets/1edd9048-a422-4539-a5cf-1872a3eb803b)

Please have a look, thanks @dingosky

**Comment by dingosky** [comment]
Thx, I'm able to reproduce the warning now and will dig into it further....

**Comment by dingosky** [comment]
Fixed in 263fc6b0 (2.3.2)
